src/gui/addnewtorrentdialog.cpp

```diff
@@ -246,53 +245,44 @@ void AddNewTorrentDialog::show(const QString &source, const BitTorrent::AddTorre
     const BitTorrent::MagnetUri magnetUri(source);
     const bool isLoaded = magnetUri.isValid()
         ? dlg->loadMagnet(magnetUri)
-        : dlg->loadTorrent(source);
+        : dlg->loadTorrentFile(source);
 
-    if (isLoaded) {
-#ifdef Q_OS_MAC
-        dlg->exec();
-#else
-        dlg->open();
-#endif
-    }
-    else {
+    if (!isLoaded) {
         delete dlg;
+        return;
```

Comment:
So now the following would be better (since there is no reason for "earlier return" now):
```c++
if (isLoaded)
    dlg->openDialog();
else
    delete dlg;
```

Reply:
updated.